File: Second Year/Second Semester/Web Programming/ExamPreparation/MockExams/2022-main-day1/JSP-Exam/src/main/java/web/exam/jspexam/model/Document.java
package web.exam.jspexam.model;

import org.json.JSONObject;

public class Document {
    private Integer id;
    private String title;
    private String templates;

    public Document(Integer id, String title, String templates) {
        this.id = id;
        this.title = title;
        this.templates = templates;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getTemplates() {
        return templates;
    }

    public void setTemplates(String templates) {
        this.templates = templates;
    }

    public JSONObject convertToJson() {
        JSONObject jsonObj = new JSONObject();

        jsonObj.put("id", id);
        jsonObj.put("title", title);
        jsonObj.put("templates", templates);

        return jsonObj;
    }
}

File: Second Year/Second Semester/Web Programming/ExamPreparation/MockExams/2022-main-day1/PHP-Exam/server/model/document.php
<?php
class Template implements JsonSerializable {
    private $id;
    private $title;
    private $templates;

    public function __construct(int $id, string $title, string $templates) {
        $this->id = $id;
        $this->title = $title;
        $this->templates = $templates;
    }

    public function getId(): int {
        return $this->id;
    }
 
    public function getTitle() : string {
        return $this->title;
    }

    public function getTemplates() : string {
        return $this->templates;
    }

    public function jsonSerialize() {
        return [
            'id' => $this->id,
            'title' => $this->title,
            'templates' => $this->templates,
        ];
    }
}
?>
